docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+:showtitle:
+:numbered:
+:toc: left
+
+//
+// The sources for the diagrams in this document are in the Oxide Google Drive
+// folder for Control Plane Architecture:
+// https://drive.google.com/open?id=1OI-QxmapK7oYGFRGp0suJdpQDft-qVAz
+//
+
+= Control plane architecture
+
+NOTE: Much of this material originally came from <<rfd48>> and <<rfd61>>.  This is now the living documentation for all the material covered here.
+
+NOTE: The RFD references in this documentation may be Oxide-internal.  Where possible, we're trying to move relevant documentation from those RFDs into docs here.
+
+== What is the control plane
+
+In software systems the terms **data plane** and **control plane** are often used to refer to the parts of the system that directly provide resources to users (the data plane) and the parts that support the configuration, control, monitoring, and operation of the system (the control plane).  Within the Oxide system, we say that the data plane comprises those parts that provide CPU resources (including both the host CPU and hypervisor software), storage resources, and network resources.  The control plane provides the APIs through which users provision, configure, and monitor these resources and the mechanisms through which these APIs are implemented.  Also part of the control plane are the APIs and facilities through which operators manage the system itself, including fault management, alerting, software updates for various components of the system, and so on.
+
+Broadly, the control plane must provide:
+
+* an externally-facing API endpoint described in <<rfd4>> through which users can provision elastic infrastructure backed by the system.  This includes APIs for compute Instances, storage, networking, as well as supporting resources like organizations, users, groups, ssh keys, tags, and so on.  This API may be used by developers directly as well as the developer console backend.  See <<rfd30>>.
```

Comment:
I find it unusual that Instance is capitalized, but I see that it comes from RFD 48 and RFD 61, but not from RFD 4.

Reply:
Yeah, we tried to use this convention for a while, where names of different API resources were capitalized.  This helped distinguish "Instance" the API resource from an instance of a control plane zone, or an SMF service, or a zillion other things.  We got inconsistent about it and mostly stopped doing it.  I just went ahead and made them all 'instance'.